src/tests/ftest/util/command_utils.py

```diff
@@ -1062,38 +1064,41 @@ def copy_configuration(self, hosts):
                     raise CommandFailure(
                         f"ERROR: Copying yaml configuration file to {result.failed_hosts}")
 
-    def verify_socket_directory(self, user, hosts):
+    def verify_socket_directory(self, user, hosts, privileged=False):
         """Verify the domain socket directory is present and owned by this user.
 
         Args:
             user (str): user to verify has ownership of the directory
             hosts (list): list of hosts on which to verify the directory exists
+            privileged (bool, optional): does the directory creation and change of ownership require
+                privileged access. Defaults to False.
 
         Raises:
             CommandFailure: if the socket directory does not exist or is not
                 owned by the user and could not be created
 
         """
-        if self.yaml is not None:
-            directory = self.get_socket_dir()
+        if not hosts or not self.verify_socket_dir or not self.yaml:
+            # No need to verify the socket directory
+            return
+
+        directory = self.get_socket_dir()
+        if directory is None:
+            # No socket directory defined in the config file
+            return
+
+        self.log.info("Verifying %s socket directory: %s", self.command, directory)
+        status, nodes = check_file_exists(hosts, directory, user)
+        if not status:
             self.log.info(
-                "Verifying %s socket directory: %s", self.command, directory)
-            status, nodes = check_file_exists(hosts, directory, user)
-            if not status:
-                self.log.info(
-                    "%s: creating socket directory %s for user %s on %s",
-                    self.command, directory, user, nodes)
-                result = create_directory(self.log, nodes, directory, user="root")
-                if not result.passed:
-                    raise CommandFailure(
-                        f"{self.command}: error creating socket directory {directory} for user "
-                        f"{user} on {result.failed_hosts}")
-                result = change_file_owner(
-                    self.log, nodes, directory, user, get_primary_group(user), user="root")
-                if not result.passed:
-                    raise CommandFailure(
-                        f"{self.command}: error setting socket directory {directory} owner for "
-                        f"user {user} on {result.failed_hosts}")
+                "Creating %s socket directory %s for user %s on %s",
+                self.command, directory, user, nodes)
+            result = create_directory(
+                self.log, nodes, directory, user=user, owner="root" if privileged else None)
```

Comment:
I think this. The owner should always be the user. But we need to run as root if the directory is privileged, or if the owner is not the current user
```suggestion
            result = create_directory(
                self.log, nodes, directory,
                user="root" if (privileged or user != getuser()) else None,
                owner=user)
```

Reply:
Actually, I think this.
```suggestion
            result = create_directory(
	                self.log, nodes, directory,
	                user="root" if privileged else user,
	                owner=user)
```

Reply:
Thanks